qa/tasks/deepsea.py

```diff
@@ -142,26 +104,34 @@ def make_install(self):
             '--no-recommends',
             run.Raw('$(rpmspec --requires -q DeepSea/deepsea.spec.in 2>/dev/null)')
             ])
-        self.setup_salt()
-
-    def setup_salt(self):
-        self.log.info("listing minion keys...")
-        self.salt.master_remote.run(args = ['sudo', 'salt-key', '-L'])
 
-        self.log.info("iterating over all the test nodes...")
+    def __purge_osds(self):
+        # FIXME: purge osds only on nodes that have osd role
         for _remote in self.ctx.cluster.remotes.iterkeys():
-            self.log.info("minion configuration for {}".format(_remote.hostname))
-            _remote.run(args = ['sudo', 'systemctl', 'status',
-                'salt-minion.service'])
-            _remote.run(args = ['sudo', 'cat', '/etc/salt/minion_id'])
-            _remote.run(args = ['sudo', 'cat', '/etc/salt/minion.d/master.conf'])
-
-        self.salt.ping_minions()
+            self.log.info("stopping OSD services on {}"
+                .format(_remote.hostname))
+            _remote.run(args=[
+                'sudo', 'sh', '-c',
+                'systemctl stop ceph-osd.target ; sleep 10'
+                ])
+            self.log.info("unmounting OSD partitions on {}"
+                .format(_remote.hostname))
+            # bluestore XFS partition is vd?1 - unmount up to five OSDs
+            _remote.run(args=[
+                'sudo', 'sh', '-c',
+                'for f in vdb1 vdc1 vdd1 vde1 vdf1 ; do test -b /dev/$f && umount /dev/$f || true ; done'
+                ])
+            # filestore XFS partition is vd?2 - unmount up to five OSDs
+            _remote.run(args=[
+                'sudo', 'sh', '-c',
+                'for f in vdb2 vdc2 vdd2 vde2 vdf2; do test -b /dev/$f && umount /dev/$f || true ; done'
+                ])
 
     def setup(self):
         super(DeepSea, self).setup()
+        log.info("DeepSea task setup...")
```

Comment:
Isn't this should be in _setup_ method of base Task class, like:

```
log.info("%s task setup..." % type(self).__name__)
```

Same for the _begin_ method.

Reply:
@kshtsk This is intentional. The log message in base Task class is already there. I'm adding a separate INFO-level log message to the subclass so I can easily see from the log which commands are coming from the base Task class, and which ones from the subclass.

Reply:
I don't see any log instructions in class https://github.com/ceph/teuthology/blob/master/teuthology/task/__init__.py
For _begin_ or _setup_ methods.

I understand your intention, but it looks overloading to add such log info in each task. Probably we can discuss it with upstream and use _log.debug_ there.

Reply:
Noted - thanks for looking that up.

Still, I realized I don't care about the stuff that the base Task class does for these methods, so I don't require any log message there.